ocs_ci/ocs/node.py

```diff
@@ -1551,53 +1551,72 @@ def taint_nodes(nodes, taint_label=None):
             log.info(f"{node} was not tainted - {e}")
 
 
+def has_taint(node_obj, taint):
+    """
+    Check if a specific node has a given taint.
+
+    Args:
+        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object
+        taint (str): The taint key to check.
+
+    Returns:
+        bool: True if the node has the taint, False otherwise.
+
+    """
+    taints = node_obj.get().get("spec", {}).get("taints", [])
+    return any(t.get("key") in taint for t in taints)
+
+
 def check_taint_on_nodes(taint=None):
```

Comment:
Can these 2 functions be merged?

Reply:
The idea is to separate the functions to make it cleaner. Also, we use the function `has_taint` in the function `untaint_nodes`: https://github.com/red-hat-storage/ocs-ci/pull/11346/files#diff-1bba9ad83e489f5734ed993cec2d17018880017b2c2718e45607dcff3009f588R1611. Maybe I should mention it in the description.